base/compiler/cicache.jl

```diff
@@ -73,7 +101,7 @@ function setindex!(wvc::WorldView{InternalCodeCache}, ci::CodeInstance, mi::Meth
 end
 
 function code_cache(interp::AbstractInterpreter)
```

Comment:
Should this now be `code_cache(interp:: NativeInterpreter, ...)`? Otherwise foreign interp will silently leak information into the native cache.

Reply:
Sure, we can force external absint to explicitly declare their cache